### PR TITLE
Fix link to "Entropy (information theory)" Wikipedia entry

### LINKED ISSUE #2565: [Bug]: Link to "Entropy (information theory)" Wikipedia entry in coding-introduction/further-reading doesn't render properly
### Description of the bug

This affects both the production[0] and development[1] sites.

expected behaviour:
* the link should render as `Entropy and information theory`
* its URL should be `https://en.wikipedia.org/wiki/Entropy_(information_theory)`

actual behaviour:
* the link renders as `Entropy and information theory)` //  unexpected trailing parenthesis
* the URL is incorrect: `https://en.wikipedia.org/wiki/Entropy_(information_theory` // the URL's closing parenthesis is missing

I am attaching a screenshot taken of the dev site earlier today.

[0] https://www.csfieldguide.org.nz/en/chapters/coding-introduction/further-reading/
[1] https://cs-field-guide-dev.csse.canterbury.ac.nz/en/chapters/coding-introduction/further-reading/

### Steps To Reproduce

1. go to <https://www.csfieldguide.org.nz/en/chapters/coding-introduction/further-reading/#useful-links> (Prod) or <https://cs-field-guide-dev.csse.canterbury.ac.nz/en/chapters/coding-introduction/further-reading/#useful-links> (Dev).
2. Observe the second resource listed - `Entropy and information theory)` - is incorrectly rendered, and points to an incorrect entry, per the description above.

### What device/software are you seeing this problem on?

macos 14.4 on Safari 17.4 and Chrome 123.0.6312.59
iOS 17.4/Safari

screenshot (Dev site):

trailing_paren-230324

### Additional Information

Tentative analysis:

Newer Markdown implementations (e.g. GitHub's as of this writing) properly render an inline link[0] whose link destination[1] includes parenthesis. Older ones fail, and, as a result, what should be rendered in HTML as:

href="https://en.wikipedia.org/wiki/Entropy_(information_theory)"

is instead rendered as:

href="https://en.wikipedia.org/wiki/Entropy_(information_theory"

(closing parenthesis elided/wrong URL)

Wrapping the link destination within angle brackets solves that issue and should be forward compatible. Percent-encoding the inner parens (w/ %28 and %29, respectively) works as well.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mckzm**
Make "Entropy and information theory" inline link compatible with older markdown implementations.

fixes #2565

Newer Markdown implementations (e.g. GitHub's as of this writing) properly render an inline link[0] whose link destination[1] includes parenthesis. Older ones fail, and, as a result, what should be rendered as:

href="https://en.wikipedia.org/wiki/Entropy_(information_theory)"

is instead rendered as:

href="https://en.wikipedia.org/wiki/Entropy_(information_theory"

(closing parenthesis elided/wrong URL)

Wrapping the link destination within angle brackets solves that issue in a way that should be forward-compatible.

[0] https://spec.commonmark.org/0.31.2/#inline-link
[1] https://spec.commonmark.org/0.31.2/#link-destination

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request:

### Checklist

*Change the space in the box to an `x` for those you have completed.
You can also fill these out after creating the pull request.
If you're unsure about any of them, don't hesitate to ask.
We're here to help!
This is simply a reminder of what we are going to look for before merging your change.*

- [ ] I have read the contribution guidelines.
- [ ] I have linked any relevant existing issues/suggestions in the description above (include `#???` in your description to reference an issue, where `???` is the issue number).
- [ ] The pull request is requesting a merge into the `develop` branch.
- [ ] I have added necessary documentation (if appropriate).
- [ ] If I've added/modified/deleted a third-party system, I've updated the relevant license files.
- [ ] I have added myself to the 'Community Contributors' section of `csfieldguide/templates/appendices/contributors.html`

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc.
Feel free to add any images that might be helpful to understand the initial problem/solution.

### CODE REVIEW

**Comment by jimbonothing64** [comment]
Thanks @mckzm  for spotting and documenting this.

Unfortunately, when visiting https://cs-field-guide.localhost/en/chapters/coding-introduction/further-reading/, I get a `TemplateSyntaxError` :
```
Could not parse the remainder: '<https://en.wikipedia.org/wiki/Entropy_(information_theory' from '<https://en.wikipedia.org/wiki/Entropy_(information_theory'
```
Here is line 14 (offending line) of the template:
```
14: 	<li><a href="{% url <https://en.wikipedia.org/wiki/Entropy_(information_theory %}" target="_blank">Entropy and information theory</a>&gt;)</li>
```

This could be a bug in [verto](https://github.com/uccser/verto), our markdown renderer, where it doesn't support this feature of markdown correctly. It's worth investigating.

**Comment by jimbonothing64** [comment]
I've added a test to [verto](https://github.com/uccser/verto). It is a bug where links in <> are not rendered correctly. Your change *should* work, so once verto is fixed to support this markdown feature, I will approve your pull.

@mckzm, in the meantime, could you please add yourself to the to the 'Community Contributors' section of csfieldguide/templates/appendices/contributors.html in this commit?

**Comment by mckzm** [comment]
Thanks @jimbonothing64.

I have just added myself to the tail of the 'Community Contributors'  list and update the PR.

I've found 5 more instances of the same "dangling paren" problem in various links - would you prefer me to file one issue & PR/instance, or group them all into a single one?

**Comment by jimbonothing64** [comment]
Apologies for the delayed response. Your issue and pull request have raised some very interesting points and uncovered other interesting issues!

At some point, verto should be updated to better support links; fixing verto requires more work/time than I'm can dedicate to this problem.

@mckzm, we should move to your simpler solution, from https://github.com/uccser/cs-field-guide/issues/2565: replacing `(` and `)` with `%28` and `%29`, respectively. 

Could you please update your pull to replacing both the initial link and also the extra 5 links you discovered?

Thanks heaps!

**Comment by mckzm** [comment]
@jimbonothing64 all done :-)

File: csfieldguide/chapters/content/en/software-engineering/sections/further-reading.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 - [Wikipedia &ndash; Software crisis](https://en.wikipedia.org/wiki/Software_crisis)
 - [IEEE &ndash; Why software fails](http://spectrum.ieee.org/computing/software/why-software-fails)
 - [Wikipedia &ndash; Software design](https://en.wikipedia.org/wiki/Software_design)
-- [Wikipedia &ndash; Abstraction](https://en.wikipedia.org/wiki/Abstraction_(computer_science))
+- [Wikipedia &ndash; Abstraction](https://en.wikipedia.org/wiki/Abstraction_&28computer_science%29)
```
Comment (jimbonothing64) [inline csfieldguide/chapters/content/en/software-engineering/sections/further-reading.md:9]: should be `%28` here, not `%29`

**Review by jimbonothing64 — CHANGES_REQUESTED** [review]
Thanks so much! One link has a typo, once that's fixed I'll merge your changes in. Thanks again!

File: csfieldguide/chapters/content/en/software-engineering/sections/further-reading.md (hunk 1 above)
Comment (mckzm) [reply csfieldguide/chapters/content/en/software-engineering/sections/further-reading.md]: Ooops sorry about that - now fixed!

**Review by jimbonothing64 — APPROVED** [review]
Thanks!

**Review by jimbonothing64 — APPROVED** [review]

**Pull request merged**